### PR TITLE
feat(geometric): Add lineRotate method to geometric module (geometric v2.4.0)

### PR DESCRIPTION
**Pull request by tmcw**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/HarryStevens/geometric/blob/master/src/lines/lineRotate.js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## geometric ([<kbd>unpkg</kbd>](https://unpkg.com/browse/geometric@latest/))
was missing the following properties:
1. polygonScaleX
1. polygonScaleY
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 074386f7496dd6be51e5338e59e779b3d9f2f268
</p>

**Review by lindapaiste — APPROVED** [review]
A lot more than 2 weeks, sorry!

Your types for the new functions (`lineRotate`, `polygonScaleX`, and `polygonScaleY`) look correct and should be fine to merge.  There's a minor issue in a comment, but nothing inaccurate in the types.   Thank you for your contribution!

File: types/geometric/index.d.ts [hunk 1]
```diff
@@ -7,194 +7,218 @@ export as namespace geometric;
...
-     */
-    export function lineMidpoint(line: Line): Point;
+/**
+ * Returns the midpoint of a line.
+ */
+export function lineMidpoint(line: Line): Point;
+
+/**
+ * Returns the vertices resulting from rotating a line about an origin by an angle in degrees. If origin is not
+ * specified, the origin defaults to [0, 0].
```
Comment (lindapaiste) [inline types/geometric/index.d.ts:68]: The default origin here should be the midpoint
> Returns the coordinates resulting from rotating a line about an origin by an angle in degrees. If origin is not specified, the origin defaults to the [midpoint](https://github.com/HarryStevens/geometric#lineMidpoint) of the line.

https://github.com/HarryStevens/geometric#lineRotate

**Comment by lindapaiste** [comment]
Ready to merge

**Pull request merged**